tools/c7n_azure/tests/test_storage.py

```diff
@@ -522,20 +524,31 @@ def test_storage_settings_update_logging_table(self, mock_get_storage_key,
         StorageSettingsUtilities.update_logging(
             TABLE_TYPE, mock_storage_account, log_settings, session=mock_session)
 
-        mock_get_storage_key.assert_called_with(
-            'mock_resource_group', 'mock_storage_account', mock_session)
-        mock_set_table_properties.assert_called_once()
-
-    @patch('azure.storage.blob.blockblobservice.BlockBlobService.set_blob_service_properties')
-    def test_storage_settings_update_logging_blob(self, mock_set_blob_properties):
-        mock_storage_account = {
-            "resourceGroup": "mock_resource_group",
-            "name": "mock_storage_account"
-        }
-        mock_token = 'mock_token'
-        log_settings = MagicMock()
-
-        StorageSettingsUtilities.update_logging(
-            BLOB_TYPE, mock_storage_account, log_settings, token=mock_token)
-
-        mock_set_blob_properties.assert_called_once()
+    def test_storage_settings_enable_https_traffic_action(self):
+        with patch('azure.mgmt.storage.v%s.operations.'
+        '_storage_accounts_operations.StorageAccountsOperations.update'
```

Comment:
I wonder if it is necessary to patch a specific version?

Maybe something like `azure.mgmt.storage.storage_accounts.update` ?

They sort of wire these up for you here with the version automatically:
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-mgmt-storage/azure/mgmt/storage/_storage_management_client.py#L246

Reply:
Unfortunately, I think this is the best way for us to get to the actual function to `patch` since we don't have access to the StorageClient object in the test.